rpc/server.go

```diff
@@ -0,0 +1,103 @@
+package rpc
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/reflection"
+)
+
+var log logrus.FieldLogger
+
+func init() {
+	log = logrus.WithField("prefix", "rpc")
+}
+
+// Config options for the gRPC server.
+type Config struct {
+	Host     string
+	Port     string
+	CertFlag string
+	KeyFlag  string
+}
+
+// Server defining a gRPC server for the remote signer API.
+type Server struct {
+	ctx             context.Context
+	cancel          context.CancelFunc
+	host            string
+	port            string
+	listener        net.Listener
+	withCert        string
+	withKey         string
+	jwtKey          string
+	auth0Audience   string
+	credentialError error
+	grpcServer      *grpc.Server
+}
+
+// NewServer instantiates a new gRPC server.
+func NewServer(ctx context.Context, cfg *Config) *Server {
+	ctx, cancel := context.WithCancel(ctx)
+	return &Server{
+		ctx:      ctx,
+		cancel:   cancel,
+		host:     cfg.Host,
+		port:     cfg.Port,
+		withCert: cfg.CertFlag,
+		withKey:  cfg.KeyFlag,
+	}
+}
+
+// Start the gRPC server.
+func (s *Server) Start() {
+	address := fmt.Sprintf("%s:%s", s.host, s.port)
+	lis, err := net.Listen("tcp", address)
+	if err != nil {
+		log.Errorf("Could not listen to port in Start() %s: %v", address, err)
+	}
+	s.listener = lis
+	log.WithField("address", address).Info("RPC rpc listening on port")
+
+	if s.withCert != "" && s.withKey != "" {
+		creds, err := credentials.NewServerTLSFromFile(s.withCert, s.withKey)
+		if err != nil {
+			log.Errorf("Could not load TLS keys: %s", err)
+			s.credentialError = err
+		}
+		opts = append(opts, grpc.Creds(creds))
+	} else {
+		log.Warn("You are using an insecure gRPC connection! Provide a certificate and key to connect securely")
```

Comment:
We must force a secure connection here, it should fatal if there are no tls creds